force-app/main/default/classes/AFFL_MultiRecordTypeMapper.cls

```diff
@@ -78,37 +91,53 @@ public virtual with sharing class AFFL_MultiRecordTypeMapper {
      ********************************************************************************************************/
     public AFFL_MultiRecordTypeMapper() {
         //List of Contact field names.
+        //NOTE: Target for Refactor to use UTIL_Describe in the future. Changing this is high-risk and disruptive
         List<SObjectField> contactFields = Contact.sObjectType.getDescribe().fields.getMap().values();
-
         //Map of Contact field labels to Contact field Names.
-        contactFieldAPINameByLabel = getContactLabelsNamesMap(contactFields);
-
+        this.contactFieldAPINameByLabel = this.getContactLabelsNamesMap(contactFields);
         //List of primary affiliation lookup fields names.
-        List<String> contactFieldNames = contactFieldAPINameByLabel.values();
-
-        //Affiliation Account-record-type-to-Contact-field mappings.
-        populateMaps();
+        List<String> contactFieldNames = this.contactFieldAPINameByLabel.values();
 
+        //Get and cache account record type info by name.
+        //NOTE: Consider refactor of this high-risk area
+        AFFL_MultiRecordTypeMapper.accountRecordTypeInfosByName = SObjectType.Account.getRecordTypeInfosByName();
+        AFFL_MultiRecordTypeMapper.accountRecordTypeInfosByDeveloperName = SObjectType.Account.getRecordTypeInfosByDeveloperName();
         // Map of Account RecordType Label To API
-        accRecTypeLabelToAPI = getAccRecTypeLabelToAPIMap();
-
+        //NOTE: Consider a refactor relative to case sensitivity
+        this.accRecTypeLabelToAPI = this.getAccRecTypeLabelToAPIMap();
+
+        //Note: Given the supportable number of records for affiliation mappings, having multiple small loops
+        //is an acceptable and negligible performance hit so we can apply a separation of concerns for future refactoring.
+
+        //Get the affiliation mappings we have available
+        List<Affl_Mappings__c> affiliationMappings = this.getPopulatedAffiliationMappings();
+        //Make a map of the raw account record types to contact fields
+        this.primaryAfflFieldInMappingsByAccRecType = this.getRawContactFieldsByAccountRecordType(affiliationMappings);
+        //Get all mapped account record types
+        this.mappedAccountRecordTypes = this.getRawAccountRecordTypes(affiliationMappings);
+        //List of Valid Account Record types in mappings
+        //NOTE: Consider a refactor relative to case sensitivity. This is a high-risk area
+        this.validAccRecordTypesInMappings = this.getValidAccRecordTypesInMappings(this.mappedAccountRecordTypes);
+        //Get a map of account record type labels to Primary Affiliation Mapping
+        this.primaryAffiliationMappingsByAccountRecordTypeName = this.getPrimaryAffiliationMappingsByAccountRecordTypeName(
+            affiliationMappings,
+            this.validAccRecordTypesInMappings,
+            AFFL_MultiRecordTypeMapper.accountRecordTypeInfosByDeveloperName
```

Comment:
While that may be true, the refactor was moving us into a position where it would not necessarily have to be static. It also improves the readability and testability of the method as I no longer have to attempt to adjust the static variable for the method to resolve.

This is a step toward using object-oriented principles over scripting/procedural ones.